platform/src/main/java/org/stellar/anchor/platform/configurator/ConfigManager.java

```diff
@@ -0,0 +1,192 @@
+  }
+
+  void sanitize(ConfigMap configMap) {
+    SecretManager.getInstance()
+        .secretVars
+        .forEach(
+            var -> {
+              if (configMap.get(var) != null) {
+                Log.warnF(
+                    "Possible secret leak of config[{}]. Please remove the secret from configuration.",
+                    var);
+                configMap.remove(var);
+              }
+            });
+  }
+
+  void sendToSpring(
+      ConfigurableApplicationContext applicationContext,
+      ConfigMap config,
+      List<SpringConfigAdapter> adapters)
+      throws InvalidConfigException {
+    Properties props = new Properties();
+    props.putAll(config.map());
+
+    // Set Platform configurations
+    PropertiesPropertySource pps = new PropertiesPropertySource("config", props);
+    applicationContext.getEnvironment().getPropertySources().addFirst(pps);
+
+    for (SpringConfigAdapter adapter : adapters) {
+      adapter.sendToSpring(applicationContext, config);
+    }
+  }
+
+  ConfigMap processConfigurations(ConfigurableApplicationContext applicationContext)
+      throws IOException, InvalidConfigException {
+    info("reading default configuration values");
+    // Load default values
+    ConfigMap defaultConfig = loadDefaultConfig();
+    ConfigMap config = defaultConfig;
+
+    infoF("default configuration version={}", defaultConfig.getVersion());
+    // Check if default config is consistent with the definition
+    ConfigReader configReader = new ConfigReader(defaultConfig.getVersion());
+    info("validating default configuration values");
+    configReader.validate(defaultConfig);
+
+    // Read the configuration from the YAML file specified by the STELLAR_ANCHOR_CONFIG system
+    // environment variable.
+    Resource configFileResource = getConfigFileAsResource(applicationContext);
+    if (configFileResource != null) {
+      infoF("reading configuration file from {}", configFileResource.getURL());
+      ConfigMap yamlConfig = loadConfig(configFileResource, FILE);
+      config.merge(applyReaders(defaultConfig, yamlConfig));
+    }
+
+    // Read and process the environment variable
+    ConfigMap envConfig = loadConfigFromEnv(defaultConfig.getVersion());
+    if (envConfig != null) {
+      info("Processing system environment variables");
+      config.merge(applyReaders(defaultConfig, envConfig));
+    }
+
+    return config;
+  }
+
+  ConfigMap applyReaders(ConfigMap defaultConfig, ConfigMap config) throws InvalidConfigException {
+    int configVersion = config.getVersion();
+    int latestVersion = defaultConfig.getVersion();
+
+    ConfigReader configReader = new ConfigReader(configVersion);
+    configReader.validate(config);
+
+    for (int version = configVersion + 1; version <= latestVersion; version++) {
+      infoF("Applying configuration version {}", version);
+      configReader = new ConfigReader(version);
+      config = configReader.readFrom(config);
+    }
+
+    return config;
+  }
```

Comment:
The object passed from the reference `config` is actually not changed because line#128 `configReader.readFrom()` returns a new object. I tried to use a new variable but ends up make it less readable. 

I will add a comment before `configReader.readFrom` statement to describe the `readFrom` call returns a new object.

Reply:
Gotcha! My understanding was wrong then. I've tested it in http://tpcg.io/_480B40 and confirmed you're right 👍.